internal/pkg/cli/deploy/workload.go

```diff
@@ -365,9 +364,14 @@ func (d *workloadDeployer) uploadContainerImages(out *UploadArtifactsOutput) err
 	if len(buildArgsPerContainer) == 0 {
 		return nil
 	}
+	uri, err := d.repository.Login()
+	if err != nil {
+		return fmt.Errorf("login to docker: %w", err)
```

Comment:
nit: since it is not necessarily docker hub!
```suggestion
		return fmt.Errorf("login to image repository: %w", err)
```